build.sh

```diff
@@ -71,6 +72,23 @@ function hasArg {
     (( ${NUMARGS} != 0 )) && (echo " ${ARGS} " | grep -q " $1 ")
 }
 
+function cmakeArgs {
```

Comment:
I moved the cmakeArgs function into the validArgs function since that makes sense. However, the validArgs function is very simple and easy to read in its current form. The cmakeArgs function is doing more complicated parsing which is why I separated the functions. I figured this approach maintained better readability.

Reply:
Makes sense, and looks good

Thank you